superset-frontend/src/explore/reducers/datasourcesReducer.ts

```diff
@@ -0,0 +1,37 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { Dataset } from '@superset-ui/chart-controls';
+import { getDatasourceUid } from 'src/utils/getDatasourceUid';
+import {
+  AnyDatasourcesAction,
+  SET_DATASOURCE,
+} from '../actions/datasourcesActions';
+
+export default function datasourcesReducer(
+  datasources: { [key: string]: Dataset },
```

Comment:
Should I change it now? Or just add a todo when those new datasource types are introduced?

Reply:
Either way would work. Using the intersection type may cause some typescript errors because you'll need to resolve the difference in the schema of the datasources, which is something that we're working on now in a feature branch. It may be easiest to leave it as a todo, honestly.

Reply:
Added a todo